Dockerfile

```
@@ -1,17 +1,27 @@
-FROM node:14-slim
+FROM node:14-slim as build
 
 LABEL maintainer="Orta Therox"
 LABEL "com.github.actions.name"="Danger JS Action"
 LABEL "com.github.actions.description"="Runs JavaScript/TypeScript Dangerfiles"
 LABEL "com.github.actions.icon"="zap"
 LABEL "com.github.actions.color"="blue"
 
-RUN mkdir -p /usr/src/danger
-COPY . /usr/src/danger
-RUN cd /usr/src/danger && \
-  yarn && \
-  yarn run build:fast && \
-  chmod +x distribution/commands/danger.js && \
-  ln -s $(pwd)/distribution/commands/danger.js /usr/bin/danger
+WORKDIR /usr/src/danger
+COPY package.json yarn.lock ./
+RUN yarn install
+COPY . .
+RUN yarn run build:fast
+RUN yarn remove 'typescript' --dev && yarn add 'typescript@^4.5.5'
```

```suggestion
RUN yarn remove 'typescript' --dev && yarn add 'typescript'
```

I'm going to switch this to a versionless version - because the API we use from the lib will never break, and then folks don't need to think about whether they can use specific TS features

---

that is bad. typescript breaks even in minor updates. simple example `catch (e)` go changed to `catch (e: unknown)` somewhere in 4.5 -> 4.9.

how about using this tool specifically for this purpose:
- https://github.com/glensc/yarn-install-dev